### PR TITLE
Recipe of BeGUI (Need the license)

### PR DESCRIPTION
**Pull request by arnavbhatt288**
Recipe of BeGUI 1.4.2 (doesn't build properly).
Need the license.
Edit: I don't think it is necessary but I am GCI student.

### CODE REVIEW

File: haiku-apps/begui/begui-1.4.2.recipe [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+PROVIDES="
+	begui = $portVersion
+	app:PrefsEDgar = $portVersion
+	app:ResEDfar = $portVersion
+	app:Resourceful = $portVersion
+	lib:libprefs = $portVersion
+	lib:libBeGUI = $portVersion
+	"
+REQUIRES="
+        haiku
```
Comment (diversys) [inline haiku-apps/begui/begui-1.4.2.recipe]: Use tabs instead of spaces here and everywhere below.

File: haiku-apps/begui/begui-1.4.2.recipe [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+COPYRIGHT="1999 Ed Musgrove"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="07fef544ab398c73404badafb63aed75cb057917"
+SOURCE_URI="https://github.com/HaikuArchives/BeGUI/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="8438cca9cdaf71722f996fce6efc68f4aae99657b99c98d88202df97ed29b284"
+SOURCE_FILENAME="BeGUI-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="BeGUI-$srcGitRev"
+
+ARCHITECTURES="x86 x86_gcc2"
```
Comment (extrowerk) [inline haiku-apps/begui/begui-1.4.2.recipe]: It should be:
ARCHITECTURES="x86_gcc2 x86"

**Comment by extrowerk** [comment]
For further inspiration: http://0x0.st/sUq3.recipe
and http://0x0.st/sUqg.patchse

Be warned, it is not a proper patchset.

**Comment by arnavbhatt288** [comment]
Uh oh, I putted MIT license for testing recipe and forgot to remove it.
I don't know the license of this app so I must replace this immediately.

**Comment by arnavbhatt288** [comment]
> For further inspiration: http://0x0.st/sUq3.recipe
> and http://0x0.st/sUqg.patchse
> 
> Be warned, it is not a proper patchset, and while it builds on x86_/4 it complains about missing symbol. But at least you don't have to fight too mach with getting it to built.
> 
> Have fun with hacking.

Thank you for helping.
I am not in patchset so can you tell me what is the errors or flaws I need to fix?

**Comment by arnavbhatt288** [comment]
@extrowerk I am getting problems. The demo apps are crashing.

**Comment by extrowerk** [comment]
Provide us your current recipe so we check it.

**Comment by arnavbhatt288** [comment]
> Provide us your current recipe so we can check it.

**Comment by arnavbhatt288** [comment]
> Provide us your current recipe so we can check it.

**Comment by arnavbhatt288** [comment]
@extrowerk Now see the file

**Comment by extrowerk** [comment]
Patch is missing. You have to create a file in the patches subfolder (you have to create it too).

**Comment by arnavbhatt288** [comment]
@extrowerk I know it is missing. But with patches the file builds properly but the demo apps crashes. So I tried to edit it but I always got this error "Line 144 is broken"

**Comment by arnavbhatt288** [comment]
@extrowerk Here is new patch. The demo files crashes,

**Comment by scottmc** [comment]
If it's building the apps/libs, but crashing, let's mark it as broken on all platforms.  I'm unsure of the patch.  I don't think you need the "/ No newline at end file " lines.  Also, you have an ifdef without an argument, I've not seen that before. Perhaps try commenting it out using /* and */.
For the parts of the patch that you are sure of such as the makefile, let's get that fixed on the HaikuArchives repo.  Usually we just patch here for ports where we can't get our patches upstreamed in a timely manner, but for HaikuArchives we can get those add fairly quickly.  But make sure it's working first.

**Comment by arnavbhatt288** [comment]
@extrowerk @scottmc Got the problem. The apps are expecting that the libs are on same directory and not in other one. If the libs isn't on same directory, then the apps crashes (except Resourceful wghich doesn't require any libs). But I got another problem, the PrefsEDgar doesn't run, it opens but it doesn't show any window.

**Comment by arnavbhatt288** [comment]
@extrowerk @scottmc Fixed the issue :)
The PrefsEDgar doesn't show window. It just do something at background and that's it.

**Comment by scottmc** [comment]
If there isn't one already, please open an issue on the HaikuArchives repo for this regarding the license.

**Comment by arnavbhatt288** [comment]
@scottmc Now what can you about this? :)

File: haiku-apps/begui/begui-1.4.2.recipe [hunk 3]
```diff
@@ -0,0 +1,74 @@
+SUMMARY="A simple to use programmer's tool for rapid development of a program"
+DESCRIPTION="BeGUI is a simple to use programmer's tool which allows for rapid development \
+of a program. It is sensitive to font sizes, style and families; has a lot of \
+color control preferences and had a window class which is self-sizing and \
+recalls it's position from instance to instance"
+HOMEPAGE="https://github.com/HaikuArchives/BeGUI"
+COPYRIGHT="1999 Ed Musgrove"
+LICENSE=""
```
Comment (Begasus) [inline haiku-apps/begui/begui-1.4.2.recipe]: Put in Unknown:
```diff
-LICENSE=""
+LICENSE="Unknown"
```

File: haiku-apps/begui/begui-1.4.2.recipe [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+REVISION="1"
+srcGitRev="07fef544ab398c73404badafb63aed75cb057917"
+SOURCE_URI="https://github.com/HaikuArchives/BeGUI/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="8438cca9cdaf71722f996fce6efc68f4aae99657b99c98d88202df97ed29b284"
+SOURCE_FILENAME="BeGUI-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="BeGUI-$srcGitRev"
+PATCHES="begui-$portVersion.patchset"
+
+ARCHITECTURES="x86 x86_gcc2 x86_64"
```
Comment (Begasus) [inline haiku-apps/begui/begui-1.4.2.recipe]: As the recipe works but the port isn't really suitable, disable arch's:
```diff
-ARCHITECTURES="x86 x86_gcc2 x86_64"
+ARCHITECTURES="x?86_gcc2 ?x86_64"
```

File: haiku-apps/begui/patches/begui-1.4.2.patchset [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
++}
+diff --git a/BeGUIlib/specificColorWindow.cpp b/BeGUIlib/specificColorWindow.cpp
+index 88f3c34..498b331 100644
+--- a/BeGUIlib/specificColorWindow.cpp
++++ b/BeGUIlib/specificColorWindow.cpp
+@@ -676,6 +676,7 @@ SpecificColorWindow	::	MessageReceived(	BMessage * paramMessage)
+     		rgb_color * pRGBColor;
+     		rgb_color color;   		
+     		int32 numRGBBytes = sizeof(struct rgb_color);
++    	#ifdef
```
Comment (Begasus) [inline haiku-apps/begui/patches/begui-1.4.2.patchset]: You should ifdef SOMETHING here, same as the one before, but it should fit some better description/pointer on why it's needed?

Comment (Begasus) [reply haiku-apps/begui/patches/begui-1.4.2.patchset:76]: On that note, builds ok without this #ifdef's ?

File: haiku-apps/begui/patches/begui-1.4.2.patchset [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+diff --git a/PrefsEDgar/PrefsEDgar.makefile b/PrefsEDgar/PrefsEDgar.makefile
+index ff76507..61e115b 100644
+--- a/PrefsEDgar/PrefsEDgar.makefile
++++ b/PrefsEDgar/PrefsEDgar.makefile
+@@ -64,7 +64,7 @@ RSRCS =
+ #	- 	if your library does not follow the standard library naming scheme,
+ #		you need to specify the path to the library and it's name.
+ #		(e.g. for mylib.a, specify "mylib.a" or "path/mylib.a")
+-LIBS = be /boot/home/config/non-packaged/lib/libBeGUI.so prefs
++LIBS = be BeGUI prefs $(STDCPPLIBS)
```
Comment (Begasus) [inline haiku-apps/begui/patches/begui-1.4.2.patchset:104]: Is $(STDCPPLIBS) needed in here (and elsewhere)?

**Pull request closed without merging**